controls/raddocking/how-to/add-menu-items-to-the-radpanes-menu.md

```diff
@@ -14,136 +14,220 @@ The purpose of this tutorial is to show you how to customize the __RadPane__ men
+						e.CanExecute = false;
+					}
+				}
+			}
 		}
 	{{endregion}}
 
-1. Your menu command is ready and you have a custom __DataTemplate__, which is set to the __ContextMenuTemplate__ property. The next step is to register your custom command by using the __Telerik.Windows.Controls.CommandManager__ class. You can do that in the following manner:
+1. Your menu command is ready and you have a custom __DataTemplate__, which is set to the __ContextMenuTemplate__ property. The next step is to register your custom command by using the __Telerik.Windows.Controls.CommandManager__ class in the __CustomRadDockingCommands__ constructor class. __Example 4__ show the final structure of __CustomRadDockingCommands__ class.
 	
-	#### __C#__
-
+	#### __[C#] Example 4: Register CloseAllPanesButThisCommand custom command__
 	{{region cs-raddocking-how-to-add-menu-items-to-the-radpanes-menu_3}}
-		CommandManager.RegisterClassCommandBinding(
-		    typeof(RadPaneGroup),
-		     new CommandBinding(
-		        RadDockingCommands.CloseAllPanesButThisCommand,
-		        null,
-		        null));
-	{{endregion}}
+		public class CustomRadDockingCommands
+		{
+			private static RoutedUICommand closeAllPanesButThisCommand;
+			public static RoutedUICommand CloseAllPanesButThisCommand
+			{
+				get
+				{
+					if (closeAllPanesButThisCommand == null)
+					{
+						closeAllPanesButThisCommand = new RoutedUICommand("Close all panes but this", "CloseAllPanesButThisCommand", typeof(RadDockingCommands));
+					}
+					return closeAllPanesButThisCommand;
+				}
+			}
+
+			static CustomRadDockingCommands()
+			{
+				CommandManager.RegisterClassCommandBinding(typeof(RadPaneGroup), new CommandBinding(CustomRadDockingCommands.CloseAllPanesButThisCommand, OnCloseAllPanesButThis, OnCloseAllPanesButThisCanExecute));
+			}
 
+			public static void OnCloseAllPanesButThis(object sender, ExecutedRoutedEventArgs e)
+			{
+				var pane = e.Parameter as RadPane;
+				if (pane != null)
+				{
+					var paneGroup = pane.PaneGroup;
+					if (paneGroup != null)
+					{
+						var panesToClose = paneGroup.EnumeratePanes().Where(x => !x.IsHidden && x.IsPinned);
+						foreach (var paneToClose in panesToClose)
+						{
+							if (paneToClose != pane)
+							{
+								paneToClose.IsHidden = true;
+							}
+						}
+					}
+				}
+			}
+
+			public static void OnCloseAllPanesButThisCanExecute(object sender, CanExecuteRoutedEventArgs e)
+			{
+				e.CanExecute = false;
+				var paneGroup = sender as RadPaneGroup;
+				if (paneGroup != null)
+				{
+					int childrenCount = paneGroup.EnumeratePanes().Count(x => !x.IsHidden && x.IsPinned);
+
+					if (childrenCount > 1)
+					{
+						e.CanExecute = true;
+					}
+					else
+					{
+						e.CanExecute = false;
+					}
+				}
+			}
+		}
+	{{endregion}}
 
 Run your demo. The __RadPane__'s menu should look like the snapshot below.
+
+![Customize or Remove the RadPane's Menu](images/RadDocking_HowTo_Customize_Or_Remove_The_RadPanes_Menu.png)

```

Comment:
We can add Figure 1,2 etc. to be consistent with other articles. Then we can remove the above line "The __RadPane__'s menu should look like the snapshot below."